gnomad_qc/v3/create_release/create_hgdp_tgp_subset.py

```diff
@@ -0,0 +1,821 @@
+    :return: Table keyed by sample (s) with all relationship information annotated as a struct.
+    :rtype: hl.Table
+    """
+    relatedness_ht = relatedness_ht.filter(relatedness_ht.relationship != UNRELATED)
+    relationship_struct = hl.struct(
+        kin=relatedness_ht.kin,
+        ibd0=relatedness_ht.ibd0,
+        ibd1=relatedness_ht.ibd1,
+        ibd2=relatedness_ht.ibd2,
+        relationship=relatedness_ht.relationship,
+    )
+
+    relatedness_ht_i = relatedness_ht.group_by(s=relatedness_ht.i.s).aggregate(
+        relationships=hl.agg.collect_as_set(
+            hl.struct(s=relatedness_ht.j.s, **relationship_struct)
+        )
+    )
+
+    relatedness_ht_j = relatedness_ht.group_by(s=relatedness_ht.j.s).aggregate(
+        relationships=hl.agg.collect_as_set(
+            hl.struct(s=relatedness_ht.i.s, **relationship_struct)
+        )
+    )
+
+    relatedness_ht = relatedness_ht_i.union(relatedness_ht_j)
+
+    return relatedness_ht
+
+
+def prepare_sample_annotations() -> hl.Table:
+    """
+    Load meta HT and select row and global annotations for HGDP + TGP subset.
+
+    :return: Table containing sample metadata for the subset
+    """
+
+    logger.info(
+        "Sub-setting and modifying sample QC metadata to desired globals and annotations"
+    )
+    meta_ht = meta.ht()
+    meta_ht = meta_ht.filter(meta_ht.subsets.hgdp | meta_ht.subsets.tgp)
+    meta_ht = meta_ht.select_globals(
+        global_annotation_descriptions=hl.literal(GLOBAL_SAMPLE_ANNOTATION_DICT),
+        sample_annotation_descriptions=hl.literal(SAMPLE_ANNOTATION_DICT),
+        sex_imputation_ploidy_cutoffs=meta_ht.sex_imputation_ploidy_cutoffs,
+        population_inference_pca_metrics=hl.struct(
+            n_pcs=meta_ht.population_inference_pca_metrics.n_pcs,
+            min_prob=meta_ht.population_inference_pca_metrics.min_prob,
+        ),
+        hard_filter_cutoffs=meta_ht.hard_filter_cutoffs,
+    )
+
+    relatedness_ht = relatedness.ht()
+    subset_samples = meta_ht.s.collect()
+    relatedness_ht = relatedness_ht.filter(
+        subset_samples.contains(relatedness_ht.i.s)
+        & subset_samples.contains(relatedness_ht.j.s)
+    )
+
+    relatedness_ht = get_relatedness_set_ht(relatedness_ht)
+    meta_ht = meta_ht.select(
+        bam_metrics=meta_ht.bam_metrics,
+        subsets=meta_ht.subsets.select("tgp", "hgdp"),
+        sex_imputation=meta_ht.sex_imputation.drop("is_female"),
+        sample_qc=meta_ht.sample_qc.select(*SAMPLE_QC_METRICS),
+        relatedness_inference=relatedness_ht[meta_ht.key],
+        population_inference=meta_ht.population_inference.drop(
+            "training_pop", "training_pop_all"
+        ),
+        labeled_subpop=meta_ht.project_meta.project_subpop,
+        gnomad_release=meta_ht.release,
+        high_quality=meta_ht.high_quality,
+    )
+
+    return meta_ht
+
+
+# TODO: move to common area for use in release HT code? Current version is a little different than Grace's version
```

Comment:
Agree it should be in the QC repo, I was thinking more like adding to a utils script like I did here https://github.com/broadinstitute/gnomad_qc/pull/191/files#diff-96ac2545da8cdf663336901b73a97c0c3b8769d5d7d8b1235d74690927fc5fd2. At this point Grace already put one in to the sites HT release code (`add_release_annotations` in https://github.com/broadinstitute/gnomad_qc/pull/189), so it would need modification, but I might rethink this after that PR is in. I will leave as is for now.